webots_ros2_driver/webots_ros2_driver/utils.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python
+
+# Copyright 1996-2021 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This launcher simply start Webots."""
+
+import os
+import re
+import sys
+import shutil
+import tarfile
+import functools
+import subprocess
+import urllib.request
+from pathlib import Path
+
+
+MINIMUM_VERSION_STR = 'R2021b'
+TARGET_VERSION_STR = 'R2021b'
+
+
+@functools.total_ordering
+class WebotsVersion:
+    def __init__(self, version):
+        self.version = version
+
+        # Parse
+        parts = re.findall(r'R(\d*)([a-b]{1})(\s((revision)|(rev))\s(\d){1}){0,1}', version)
```

Comment:
Side note: this only supports two major releases per yeas (`a` and `b`) but we could also have `c`, `d`, etc.